common/include/common/common_utils.hpp

```diff
@@ -1199,49 +1255,58 @@ static inline std::vector<omatadd_param_t<scalar_t>> get_omatadd_params(
 }
 
 /**
- * @fn get_matcopy_batch_params
- * @brief Returns a vector containing the matcopy_batch benchmark parameters,
- * either read from a file according to the command - line args, or the default
+ * @fn get_omatadd_batch_params
+ * @brief Returns a vector containing the omatadd_batch benchmark parameters,
+ * either read from a file according to the command-line args, or the default
  * ones.
  */
 template <typename scalar_t>
-static inline std::vector<matcopy_batch_param_t<scalar_t>>
-get_matcopy_batch_params(Args& args) {
+static inline std::vector<omatadd_batch_param_t<scalar_t>>
+get_omatadd_batch_params(Args& args) {
   if (args.csv_param.empty()) {
     warning_no_csv();
-    std::vector<matcopy_batch_param_t<scalar_t>> matcopy_batch_default;
+    std::vector<omatadd_batch_param_t<scalar_t>> omatadd_batch_default;
     constexpr index_t dmin = 256, dmax = 8192;
     constexpr scalar_t alpha{2};
+    constexpr scalar_t beta{2};
     constexpr index_t batch_size{5};
     constexpr index_t stride_a_mul{1};
     constexpr index_t stride_b_mul{1};
-    for (char trans : {'n', 't'}) {
-      for (index_t m = dmin; m <= dmax; m *= 2) {
-        for (index_t n = dmin; n <= dmax; n *= 2) {
-          for (index_t lda_mul = 1; lda_mul < 2; ++lda_mul) {
-            for (index_t ldb_mul = 1; ldb_mul < 2; ++ldb_mul) {
-              matcopy_batch_default.push_back(
-                  std::make_tuple(trans, m, n, alpha, lda_mul, ldb_mul,
-                                  stride_a_mul, stride_b_mul, batch_size));
+    constexpr index_t stride_c_mul{1};
+    for (char trans_a : {'n', 't'}) {
+      for (char trans_b : {'n', 't'}) {
+        for (index_t m = dmin; m <= dmax; m *= 2) {
+          for (index_t n = dmin; n <= dmax; n *= 2) {
+            for (index_t lda_mul = 1; lda_mul < 2; ++lda_mul) {
+              for (index_t ldb_mul = 1; ldb_mul < 2; ++ldb_mul) {
+                for (index_t ldc_mul = 1; ldc_mul < 2; ++ldc_mul) {
+                  omatadd_batch_default.push_back(
+                      std::make_tuple(trans_a, trans_b, m, n, alpha, beta,
+                                      lda_mul, ldb_mul, ldc_mul, stride_a_mul,
+                                      stride_b_mul, stride_c_mul, batch_size));
+                }
```

Comment:
I think you need to exclude the configurations that do not fit the conditions for `ld{a,b}`. I'm getting `Invalid lda` for `bench_omatadd_batched`.

Reply:
Thanks a lot for pointing to this! Actually it was caused by a typo in the `omatadd_batched.cpp` benchmark files itself, it computed the `lda` based on `trans_b` and now it's fixed. The generated configurations would not cause any error as they only contain leading dimension multipliers(>=1) and will thus always result in valid ld relatively to the m & n parameters.